py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/s3_utils.py

```diff
@@ -144,13 +144,59 @@ def move_s3_objects(file_list: list[str], destination: str) -> None:
     cpu_count = cast(int, os.cpu_count() if os.cpu_count() is not None else 1)
     pool_size = min(32, cpu_count + 4)
 
-    logging.info(
-        "Moving %s files to %s using %d threads",
-        len(file_list),
-        destination,
-        pool_size,
+    process_logger = ProcessLogger(
+        "move_s3_objects",
+        pool_size=pool_size,
+        destination=destination,
+        file_count=len(files),
     )
+    process_logger.log_start()
 
     with ThreadPoolExecutor(max_workers=pool_size) as executor:
-        for filename in file_list:
+        for filename in files:
             executor.submit(_move_s3_object, filename, destination)
+
+    process_logger.log_complete()
+
+
+def write_parquet_file(
```

Comment:
this function was moved from `ingest.py`. its docstring describes it pretty well.